### PR TITLE
add "--param-type" option to ros2param list.

### PR DESCRIPTION
**Pull request by fujitatomoya**
address https://github.com/ros2/ros2cli/issues/569

Signed-off-by: Tomoya.Fujita <Tomoya.Fujita@sony.com>

### CODE REVIEW

File: ros2param/ros2param/verb/list.py [hunk 1]
```diff
@@ -97,7 +102,14 @@ def main(self, *, args):  # noqa: D102
                         print(f'{node_name.full_name}:')
                     response = future.result()
                     for name in sorted(response.result.names):
-                        print(f'  {name}')
+                        if args.param_type is True:
+                            resp = call_describe_parameters(
```
Comment (fujitatomoya) [inline ros2param/ros2param/verb/list.py:106]: Or maybe we can call this before with parameter_names=None, then search the parameter name which matches with 'name' from list service response.

Comment (clalancette) [reply ros2param/ros2param/verb/list.py:106]: Yeah, I think that would be better.  Doing a separate service call for every parameter is going to be expensive and possibly slow (particularly if the node is on a different machine on the network).

File: ros2param/ros2param/verb/list.py [hunk 2]
```diff
@@ -97,7 +102,14 @@ def main(self, *, args):  # noqa: D102
...
+                                node=node, node_name=node_name.full_name,
+                                parameter_names={name})
+                            param_type_str = get_parameter_type_string(resp.descriptors[0].type)
+                            print(f'  {name} [{param_type_str}]')
```
Comment (clalancette) [inline ros2param/ros2param/verb/list.py:110]: Personally, I would change this to something a little more descriptive:

```
print(f'  {name} (type: {param_type_str})')
```

**Review by clalancette — CHANGES_REQUESTED** [review]
A few things to change in here, but I like the idea.

File: ros2param/ros2param/verb/list.py (hunk 1 above)
Comment (fujitatomoya) [reply ros2param/ros2param/verb/list.py]: Done

**Comment by fujitatomoya** [comment]
@clalancette 

could you check once again?

**Comment by clalancette** [comment]
@fujitatomoya FYI, I rebased this code and did a few more fixups to make the type lookup even faster.

**Review by clalancette — COMMENTED** [review]
One thing that would be really nice to do here would be to add some "cli" tests for the `list` verb, like we have for `dump` in https://github.com/ros2/ros2cli/blob/master/ros2param/test/test_verb_dump.py .  I don't think it would be too hard to add, and would give us a bit more verification of this work.  @fujitatomoya do you have some time to look into that?

**Comment by clalancette** [comment]
> One thing that would be really nice to do here would be to add some "cli" tests for the `list` verb, like we have for `dump` in https://github.com/ros2/ros2cli/blob/master/ros2param/test/test_verb_dump.py . I don't think it would be too hard to add, and would give us a bit more verification of this work. @fujitatomoya do you have some time to look into that?

Never mind.  I missed that we had https://github.com/ros2/ros2cli/pull/389 already open, and apparently it is harder than I think since we never solved it there.

I'm going to approve this and run CI on it.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12910/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7864/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10627/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12859/)

**Comment by clalancette** [comment]
@claireyywang @mabelzhang I'd appreciate a second look here since I did some of the modifications myself.

**Review by claireyywang — APPROVED** [review]
LGTM

**Comment by clalancette** [comment]
All right, we have approvals, green CI.  I'm going to go ahead and merge, thanks @fujitatomoya !

**Comment by fujitatomoya** [comment]
@clalancette 

> I rebased this code and did a few more fixups to make the type lookup even faster.

thanks 👍 this is better!

**Pull request merged**